src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -29,16 +31,39 @@ public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorksp
             // Get the analyzer assets for installed VSIX extensions through the VSIX extension manager.
             var extensionManager = workspace.GetVsService<SVsExtensionManager, IVsExtensionManager>();
 
-            var builder = ImmutableArray.CreateBuilder<HostDiagnosticAnalyzerPackage>();
-            foreach (var extension in extensionManager.GetEnabledExtensions(AnalyzerContentTypeName))
+            // get rootfolder and shellfolder location
+            string rootFolder;
+            string shellFolder;
+            if (TryGetRootAndShellFolder(extensionManager, out shellFolder, out rootFolder))
```

Comment:
btw, if you have verified that this fix works and will unblock the diagnostic scenarios for tomorrow's build, then please go ahead with the change ignoring the comment, once you get few more signoffs/opinions. We can clean this up later if required.